src/OrbitBase/include/OrbitBase/SharedState.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_SHARED_STATE_H_
+#define ORBIT_BASE_SHARED_STATE_H_
+
+#include <absl/synchronization/mutex.h>
+
+#include <functional>
+#include <optional>
+#include <variant>
+#include <vector>
+
+namespace orbit_base {
```

Comment:
maybe use orbit_base_internal here since these are not supposed to be used outside of Promise/Capture

Reply:
Good point. I wasn't sure about our policy here. I like just adding the `_internal` suffix.

Reply:
Done.